### PR TITLE
feat: add backend|pluginmanager|statemanager ut

### PR DESCRIPTION
**Pull request by steinliber**
Signed-off-by: Meng JiaFeng <jiafeng.meng@merico.dev>

## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [x] I have added relevant tests

## Description

- Add internal/pkg/backend package `ut`
- Add internal/pkg/pluginmanager package `ut`
- Add internal/pkg/statemanager package `ut`

## Related Issues
#845

### CODE REVIEW

File: internal/pkg/backend/backend_test.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package backend_test
+
+import (
+	"github.com/devstream-io/devstream/internal/pkg/backend"
```
Comment (daniel-hutao) [inline internal/pkg/backend/backend_test.go:4]: . "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"

	"github.com/devstream-io/devstream/internal/pkg/backend"
 	"github.com/devstream-io/devstream/internal/pkg/configloader"

File: internal/pkg/backend/backend_test.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	When("use unknown backend", func() {
+		It("should return err", func() {
+			state := configloader.State{Backend: "not_exist_plug"}
+			_, err := backend.GetBackend(state)
+			Expect(err).Error().Should(HaveOccurred())
+			Expect(err.Error()).Should(Equal("the backend type < not_exist_plug > is illegal"))
+		})
+
+	})
+
```
Comment (daniel-hutao) [inline internal/pkg/backend/backend_test.go:37]: Extra blank lines

File: internal/pkg/backend/backend_test.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	When("use unknown backend", func() {
+		It("should return err", func() {
+			state := configloader.State{Backend: "not_exist_plug"}
+			_, err := backend.GetBackend(state)
+			Expect(err).Error().Should(HaveOccurred())
+			Expect(err.Error()).Should(Equal("the backend type < not_exist_plug > is illegal"))
+		})
+
```
Comment (daniel-hutao) [inline internal/pkg/backend/backend_test.go:35]: Extra blank lines

File: internal/pkg/backend/backend_test.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+		})
+	})
+
+	When("use s3 backend", func() {
+		It("should return s3 backend struct", func() {
+			state := configloader.State{Backend: "local"}
+			_, err := backend.GetBackend(state)
+			Expect(err).Error().ShouldNot(HaveOccurred())
+		})
+
```
Comment (daniel-hutao) [inline internal/pkg/backend/backend_test.go:25]: Extra blank lines

File: internal/pkg/backend/local/local_test.go [hunk 5]
```diff
@@ -0,0 +1,79 @@
+package local_test
+
+import (
+	"io/ioutil"
```
Comment (daniel-hutao) [inline internal/pkg/backend/local/local_test.go:4]: Please note the formatting of the import format and check all files.

File: internal/pkg/backend/local/local_test.go [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+		})
+
+	})
+
+	// After each test, clean file content
+	AfterEach(func() {
+		err := os.Truncate(tFileLoc, 0)
+		Expect(err).Error().ShouldNot(HaveOccurred())
+	})
+
```
Comment (daniel-hutao) [inline internal/pkg/backend/local/local_test.go:78]: Extra blank lines. Please check all files below.

**Review by daniel-hutao — CHANGES_REQUESTED** [review]

**Review by IronCore864 — APPROVED** [review]

**Comment by IronCore864** [comment]
Great work. Need another reviewer here for the new tests.

**Review by iyear — APPROVED** [review]

**Review by hxcGit — APPROVED** [review]

**Review by daniel-hutao — APPROVED** [review]

**Pull request merged**